core/src/main/java/io/confluent/kafka/schemaregistry/storage/InMemoryCache.java

```diff
@@ -174,26 +176,53 @@ public Mode mode(String subject,
   }
 
   @Override
-  public Set<String> subjects(String namespace) throws StoreException {
-    Iterator<K> allKeys = getAllKeys();
-    return findSubject(allKeys, namespace);
+  public Set<String> subjects(String subject) throws StoreException {
+    return subjects(matchingPredicate(subject));
   }
 
-  private Set<String> findSubject(Iterator<K> allKeys, String subject) {
-    Set<String> subjects = new HashSet<String>();
-    while (allKeys.hasNext()) {
-      K k = allKeys.next();
+  @Override
+  public Set<String> subjects(Predicate<String> match) throws StoreException {
```

Comment:
perhaps it's cleaner to keep the predicate out of the interface and make it an implementation detail?